packages/framework/client-logger/fluid-telemetry/src/test/containerTelemetryEndToEnd.spec.realsvc.ts

```diff
@@ -0,0 +1,241 @@
+		 */
+		public consume(event: IFluidTelemetry): void {
+			return;
+		}
+	}
+
+	before(() => {
+		tinyliciousClient = new TinyliciousClient({ connection: { port: 7070 } });
+		schema = {
+			initialObjects: {
+				sharedMap1: SharedTree,
+			},
+		};
+
+		testTelemetryConsumer = new TestTelemetryConsumer();
+		telemetryConsumerConsumeSpy = spy(testTelemetryConsumer, "consume");
+	});
+
+	beforeEach(() => {
+		// We need to reset the telemetry consumer spy for each test so the trackEvent calls
+		// from the last test don't bleed into the next one.
+		telemetryConsumerConsumeSpy.resetHistory();
+	});
+
+	it("IFluid container's 'connected' system event produces expected ContainerConnectedTelemetry using AppInsightsTelemetryConsumer", async () => {
+		const { container } = await tinyliciousClient.createContainer(schema);
+
+		const containerId = await container.attach();
+		startTelemetry({
+			container,
+			containerId,
+			consumers: [testTelemetryConsumer],
+		});
+
+		// We don't know exactly when the given container events will fire and we can't await specific events so we have to
+		// wrap the container.on(...) event handler within a promise that will be awaited at the end of the test.
+		const testCompletePromise = new Promise<void>((resolve, reject) => {
+			// We Capture and analyze telemetry after the container connected event
+			container.on("connected", () => {
+				// Obtain the calls made to the TestTelemetryConsumer consume() method
+				const actualConsumedConnectedTelemetry: ContainerConnectedTelemetry[] =
+					telemetryConsumerConsumeSpy
+						.getCalls()
+						.map((spyCall) => spyCall.args[0] as IFluidTelemetry)
+						.filter(
+							(telemetry) =>
+								telemetry.eventName === ContainerTelemetryEventNames.CONNECTED,
+						) as ContainerConnectedTelemetry[];
+
+				if (actualConsumedConnectedTelemetry.length === 0) {
+					console.log("Failed to find expected telemetry");
+					reject(
+						new Error(
+							"Expected TestTelemetryConsumer.consume() to be called alteast once with expected container telemetry but was not.",
+						),
+					);
+				}
+
+				const actualContainerTelemetry = actualConsumedConnectedTelemetry[0];
+				const expectedContainerTelemetry: ContainerConnectedTelemetry = {
+					eventName: ContainerTelemetryEventNames.CONNECTED,
+					containerId,
+					containerInstanceId: actualContainerTelemetry.containerInstanceId,
+				};
+
+				try {
+					expect(expectedContainerTelemetry).to.deep.equal(actualContainerTelemetry);
+					// We won't know what the container containerInstanceId will be but we can still check that it is defined.
+					expect(actualContainerTelemetry.containerInstanceId)
+						.to.be.a("string")
+						.with.length.above(0);
+					// This will enable the test to finally complete.
+					resolve();
+				} catch (error) {
+					reject(error); // Reject the promise with the assertion error
+				}
```

Comment:
If we move this out of the event listener as described above, we don't need the try/catch anymore.

Reply:
Updated in [commit 9b0f166](https://github.com/microsoft/FluidFramework/pull/21092/commits/9b0f16632ac3d7fc23f7e74e69c20f5d970da5b7)